### DISCUSSION TITLE
Sensitivity scenarios

### DISCUSSION THREAD

**Discussion by giaminhhoang** [JonasBchrt/snapshot-gnss-algorithms#3] [Q&A] [answered]
Hello,
I am curious about how SnapperGPS's sensitivity. Does it support coherent and non-coherent accumulations to improve sensitivity? Can it work with CN0 about 25 - 28 dB?
Besides, when multi-constellation is enabled, it looks for GPS first or the strongest satellite first?
Thank you. 
This is very nice work.
Best regards,
Minh

**Comment by JonasBchrt** [comment, marked as answer]
Hi @giaminhhoang , let me move straight to your questions:

> Does it support coherent and non-coherent accumulations to improve sensitivity?

Yes. SnapperGPS implements coherent an non-coherent integration. The implementation described in the paper as well as the backend that is currently online use non-coherent integration.

> Can it work with CN0 about 25 - 28 dB?

Honestly, we have not done an analysis of the carrier-to-noise-density ratio (CN0). Might have been interesting as a performance measure, though. However, calculating this value is not straight-forward and also not necessary in our SnapperGPS pipeline. We do not "lock" to certain satellite carrier signals. SnapperGPS does not have a tracking stage, unlike traditional receivers. We also do not need the CN0 value to decide whether a satellite shall enter the stage in which the receiver position is calculated. There is no such threshold check.
In our paper, we compare different positioning algorithms. Among them is direct position estimation (DPE, also known as collective detection or direct positioning). This one turns out to be the most robust one to weak signals, which agrees with the existing literature, which claims that DPE is (slightly) more robust in challenging scenarios than least-squares methods.
I might have a look whether the CN0 is something that I can get out of our code or not and might come back to you.

> When multi-constellation is enabled, it looks for GPS first or the strongest satellite first?

SnapperGPS performs satellite acquisition in the cloud. To benefit from the available resources, we (massively) parallelise satellite acquisition. So, several satellites and even several satellite system or several pieces of data with different timestamps might be processed at the same time rather than sequentially.
Or does your question refer to the positioning stage rather than to the acquisition stage? GPS is not a special citizen there, too.
(To be precise, SnapperGPS has an accelerated mode where it looks for GPS first and if it cannot find any GPS satellites, then skips Galileo and BeiDou, but this mode was neither used for the paper not for the backend that is currently online.)

I hope this helps, but if I did not get to the core of one of your questions, then you are invited to come back to me.

**Comment by giaminhhoang** [comment]
Thanks a lot for quick answer. It is much clearer.
My question was indeed at acquisition stage. But as you are doing acquisition on cloud, then it doesn't matter. I think if the acquisition is done on-device, Galileo seems more costly than GPS. Hence, looking for GPS first may be a good choice.
Thanks again.
